plugins/kubernetes/app/models/kubernetes/resource.rb

```diff
@@ -96,9 +97,12 @@ def update
         expire_cache
       end
 
-      # hook for subclasses to override
       def template_for_update
-        @template
+        copy = @template.deep_dup
+        if @autoscaled && spec = copy[:spec]
+          spec.delete(:replicas)
```

Comment:
should this be `copy`?

Reply:
no, it's spec.replicas

Reply:
oh, I thought this read `spec == copy[:spec]`, but nvm...